test/functional/wallet_v3_txs.py

```diff
@@ -0,0 +1,560 @@
+
+        inputs=[]
+        outputs = {self.bob.getnewaddress() : 2.0}
+        self.send_tx(self.charlie, inputs, outputs, version_a)
+
+        assert_equal(self.bob.getbalances()["mine"]["trusted"], 0)
+        assert_greater_than(self.bob.getbalances()["mine"]["untrusted_pending"], 0)
+
+        inputs = []
+        outputs = {self.alice.getnewaddress() : 1.0}
+
+        raw_tx_v2 = self.bob.createrawtransaction(inputs=inputs, outputs=outputs, version=version_b)
+
+        assert_raises_rpc_error(
+            -4,
+            "Insufficient funds",
+            self.bob.fundrawtransaction,
+            raw_tx_v2, {'include_unsafe': True}
+        )
+
+    @cleanup
+    def v2_tx_spends_confirmed_v3_tx(self, version_a, version_b):
+        self.log.info(f"Test unavailable funds when v{version_a} tx spends confirmed v{version_b} tx")
+
+        self.generate(self.nodes[2], 1)
+
+        inputs=[]
+        outputs = {self.bob.getnewaddress() : 2.0}
+        self.send_tx(self.charlie, inputs, outputs, version_a)
+
+        assert_equal(self.bob.getbalances()["mine"]["trusted"], 0)
+        assert_greater_than(self.bob.getbalances()["mine"]["untrusted_pending"], 0)
+
+        inputs = []
+        outputs = {self.alice.getnewaddress() : 1.0}
+
+        self.generate(self.nodes[2], 1)
+
+        self.send_tx(self.bob, inputs, outputs, version_b)
+
+    @cleanup
+    def v3_utxos_appear_in_listunspent(self):
+        self.log.info("Test that unconfirmed v3 utxos still appear in listunspent")
+
+        inputs=[]
+        outputs = {self.alice.getnewaddress() : 2.0}
+        parent_txid = self.send_tx(self.charlie, inputs, outputs, 3)
+        assert_equal(self.alice.listunspent(minconf=0)[0]["txid"], parent_txid)
+
+    @cleanup
+    def truc_tx_with_conflicting_sibling(self):
+        # unconfirmed v3 tx to alice & bob
+        self.log.info("Test v3 transaction with conflicting sibling")
+        self.generate(self.nodes[2], 1)
+
+        inputs=[]
+        outputs = {self.alice.getnewaddress() : 2.0, self.bob.getnewaddress() : 2.0}
+        self.send_tx(self.charlie, inputs, outputs, 3)
+        parent_txid = self.charlie.getrawmempool()[0]
+
+        # alice spends her output with a v3 transaction
+        alice_unspent = self.alice.listunspent(minconf=0)[0]
+        inputs=[{'txid' : parent_txid, 'vout' : alice_unspent['vout']},]
+        outputs = {self.alice.getnewaddress() : alice_unspent['amount'] - Decimal(0.00000120)}
+        self.send_tx(self.alice, inputs, outputs, 3)
+
+        # bob tries to spend money
+        inputs=[]
+        outputs = {self.bob.getnewaddress() : 1.999}
+        bob_tx = self.bob.createrawtransaction(inputs=inputs, outputs=outputs, version=3)
+
+        assert_raises_rpc_error(
+            -4,
+            "Insufficient funds",
+            self.bob.fundrawtransaction,
+            bob_tx, {'include_unsafe': True}
+        )
+
+    @cleanup
+    def truc_tx_with_conflicting_sibling_change(self):
```

Comment:
In 09c9f828250e3840d7557374ff568204305e2679 "test: add truc wallet tests"

I don't see how this test is materially different from `truc_tx_with_conflicting_sibling`. Based on the naming, I assume this was added in response to https://github.com/bitcoin/bitcoin/pull/32896/commits/09c9f828250e3840d7557374ff568204305e2679#r2211372152? However, this test isn't testing anything with a change output since the parent tx is from charlie.

Reply:
I've updated this test so it actually tests spending change now.